vms/avm/vm.go

```diff
@@ -421,26 +436,47 @@ func (vm *VM) Linearize(_ context.Context, stopVertexID ids.ID, toEngine chan<-
 		vm.txBackend,
 		vm.chainManager,
 		&vm.clock,
-		vm.mempool,
+		mempool,
 	)
 
 	// Invariant: The context lock is not held when calling network.IssueTx.
-	vm.network = network.New(
+	vm.network, err = network.New(
 		vm.ctx,
 		vm.parser,
 		network.NewLockedTxVerifier(
 			&vm.ctx.Lock,
 			vm.chainManager,
 		),
-		vm.mempool,
+		mempool,
 		vm.appSender,
+		vm.registerer,
+		vm.networkConfig,
 	)
+	if err != nil {
+		return fmt.Errorf("failed to initialize network: %w", err)
+	}
+
+	// Notify the network of our current peers
+	for nodeID, version := range vm.connectedPeers {
+		if err := vm.network.Connected(ctx, nodeID, version); err != nil {
```

Comment:
If we call `Connected` with the same `nodeID` twice (without calling `Disconnected`), what happens?

I believe we are handling this right (when waiting for linearization) but curious.

Reply:
The p2p SDK just adds the nodeID to a set - so the second call would be a noop.